backend-demo/README.md

```diff
@@ -0,0 +1,16 @@
+# Portal backend demo
+
+ASHPD enables writing [portal backends](https://docs.flatpak.org/en/latest/portal-api-reference.html#idm45771421251808) that implement a subset of backend interfaces.
```

Comment:
most of the stuff is already documented in https://flatpak.github.io/xdg-desktop-portal/docs/writing-a-new-backend.html, so i would point to that and keep only the bare minimum here

Reply:
Ack. I reworked the document in that direction. Happy to iterate if you think this should be trimmed further.